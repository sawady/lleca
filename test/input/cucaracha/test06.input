

fun f(x : Int) {
  putChar(x)
  putChar(10)
}

fun main() {
  f(65)
  f(66)
  f(67)
}
