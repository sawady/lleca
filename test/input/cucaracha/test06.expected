Cons(
  Function(
    f,
    Cons(
      Parameter(
        x,
        Int
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtCall(
        putChar,
        Cons(
          ExprVar(
            x
          ),
          Nil
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              10
            ),
            Nil
          )
        ),
        Nil
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              65
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                66
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              f,
              Cons(
                ExprConstNum(
                  67
                ),
                Nil
              )
            ),
            Nil
          )
        )
      )
    ),
    Nil
  )
)
